### PR TITLE
feat(Layout): add container sizing tokens

### LINKED ISSUE #8552: Add support for container tokens
We have the following size variants that exist in the system: https://github.com/carbon-design-system/carbon/issues/8370#issue-856258839

For each one, we would like to make a container token to match it to make it easier to author styles both on our end and for consumers.

The tokens should be introduced in the `carbon-components` package and the naming scheme should follow: `$size-<name>`, for example `$size-sm`, `$size-md`, etc.

### PR DESCRIPTION
**Pull request by dakahn**
closed #8552 

## Changes
Adds container sizing tokens:
$size-xs,
$size-sm,
$size-md
$size-lg,
$size-xl,
$size-2xl

## Testing
- run `yarn build` inside @carbon/layout 
- check that `/layout/scss/generated/_size.scss` was created properly

### CODE REVIEW

File: packages/layout/src/index.js [hunk 1]
```diff
@@ -163,6 +163,20 @@ export const container = [
...
+export const sizeXLarge = rem(64);
+export const size2XLarge = rem(80);
+export const sizes = {
+  xSmall: sizeXSmall,
+  small: sizeSmall,
+  medium: sizeMedium,
+  large: sizeLarge,
+  xLarge: sizeXLarge,
+  '2XLarge': size2XLarge,
+};
```
Comment (joshblack) [inline packages/layout/src/index.js]:
```suggestion
export const sizes = {
  XSmall: sizeXSmall,
  Small: sizeSmall,
  Medium: sizeMedium,
  Large: sizeLarge,
  XLarge: sizeXLarge,
  '2XLarge': size2XLarge,
};
```

Might help for consistency with the token names themselves

**Review by joshblack — APPROVED** [review]

File: packages/layout/src/index.js (hunk 1 above)
Comment (andreancardona) [reply packages/layout/src/index.js]: agreed!

**Comment by tw15egan** [comment]
Getting undefined when running `yarn build` 

Screen Shot 2021-05-17 at 1 57 32 PM

**Review by laurenmrice — APPROVED** [review]
LGTM 👍

**Comment by joshblack** [comment]
@dakahn just seems like a quick snapshot update and you're good to go!

File: packages/themes/src/white.js [hunk 2]
```diff
@@ -304,6 +304,12 @@ export {
   container03,
   container04,
   container05,
+  sizeXSmall,
```
Comment (tw15egan) [inline packages/themes/src/white.js:309]: If we are adding this to the `white` theme, does it need to be added to each theme?

File: packages/layout/tasks/build.js [hunk 3]
```diff
@@ -54,6 +55,42 @@ async function build() {
...
+ This source code is licensed under the Apache-2.0 license found in the
+ LICENSE file in the root directory of this source tree.
+`);
+        const convert = {
+          xSmall: 'xs',
+          small: 'sm',
+          medium: 'md',
+          large: 'lg',
+          xLarge: 'xl',
+          '2XLarge': '2xl',
```
Comment (tw15egan) [inline packages/layout/tasks/build.js]:
```suggestion
          XSmall: 'xs',
          Small: 'sm',
          Medium: 'md',
          Large: 'lg',
          XLarge: 'xl',
          '2XLarge': '2xl',
```

Comment (tw15egan) [reply packages/layout/tasks/build.js]: I'm still getting `$size-undefined` when running `yarn build`. Changing these to match the capitalization in `index.js` resolved the issue

**Review by tw15egan — CHANGES_REQUESTED** [review]

**Review by tw15egan — APPROVED** [review]

**Pull request merged**